js/csv.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Reads a CSV file and returns a table with the data.
+ *
+ * @param {Blob|File} file The file object to be read.
+ * @param {Function} parser The function to be used in the
+ * CSV parsing operation.
+ */
+export const readCsv = (file, parser = null) => {
+    parser = parser || parseCsvComplex;
+    return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = e => resolve(parser(e.target.result));
+        reader.onerror = e => reject(e);
+        reader.readAsText(file);
+    });
+};
+
+/**
+ * Creates the contents of the CSV file with the data given
+ * as an array of rows and the headers.
+ *
+ * @param {Array} data Array of rows with cell data.

```

Comment:
Update docstrings, it can be something that is not an array now

Reply:
Currently it still has to be an array, it just can be either an array of objects or array of an array of strings. I'll add the feature to be able to accept an array of strings or a simple object, and then update the documentation.